client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -80,6 +81,9 @@ const Checkout = React.createClass( {
 
 		if ( this.props.cart.hasLoadedFromServer && this.props.product ) {
 			this.addProductToCart();
+			if ( this.props.couponCode ) {
```

Comment:
I think we should be able to safely do it in `addProductToCart`, thus avoiding code duplication.